### DISCUSSION TITLE
How to create a global variable in the ifelse conditionals

### DISCUSSION THREAD

**Discussion by jxpeng98** [typst/typst#808] [Q&A] [answered]
I will like to write a customised function to select the text font based on the input system type. 

The function is:
```
#let select(content) = {
  if content == "mac" {
  let f = "DM Mono"
  } else {
  let f =  "DM Sans"
  }
}

#select("mac")
#text(font:f)[#lorem(5)]
```
When I call `f` in the text setting, it will return the error `unknown variable`

---
However, if I create `f` directly and call it, there is no error:
```
#let f = "DM Mono"
#text(font:f)[#lorem(5)]
```

Can anyone help to improve this function?

Many thanks!

**Comment by johannes-wolf** [comment, marked as answer]
It is not possible. You have to return your values to the global scope and set them from there:
```typst
#let select(content) = {
  return if content == "mac" {
    "DM Mono"
  } else {
    "DM Sans"
  }
}

#let f = select("mac")
#text(font: f)[#lorem(5)]
```

**Reply by PgBiel** [reply]
It is also worth noting that, if a global variable is _really_ needed, use `state`: https://typst.app/docs/reference/meta/state/

**Reply by jxpeng98** [reply]
Thanks! This really helps. I forgot to use the simplest `return`.

**Reply by PgBiel** [reply]
> Thanks! This really helps. I forgot to use the simplest `return`.

Note that not even that is needed - you can omit the `return` keyword in Typst if it is the last expression (just like in Rust)!
